### PR TITLE
Add support for Wheatfield:Blocked65.

### PR DESCRIPTION
**Pull request by TerryMathews**
## Description

Rudimentary support for Blocked65 by Wheatfield.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by TerryMathews** [comment]
Sorry for the updates. I thought I had stripped the excess keymaps out but I guess not.

File: keyboards/wheatfield/blocked65/blocked65.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  action_function_keymap(record, id, opt);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return process_record_keymap(keycode, record);
+}
+
+void led_set_user(uint8_t usb_led) {
+  led_set_keymap(usb_led);
+}
```
Comment (drashna) [inline keyboards/wheatfield/blocked65/blocked65.c]: None of this should be needed.
```suggestion
```

File: keyboards/wheatfield/blocked65/config.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (drashna) [inline keyboards/wheatfield/blocked65/config.h]: This matches the default, and can be removed. 

```suggestion
```

File: keyboards/wheatfield/blocked65/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+/* number of backlight levels */
+
+#define BACKLIGHT_PIN B6
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 6
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/wheatfield/blocked65/config.h]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/wheatfield/blocked65/config.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+#define RGB_DI_PIN E2
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 72
```
Comment (drashna) [inline keyboards/wheatfield/blocked65/config.h]: Yikes!

Does this have external power?  

If not, you may want to limit the max brightness so you don't casue power draw issues.

File: keyboards/wheatfield/blocked65/rules.mk [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
```
Comment (drashna) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  _RAISE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_QWERTY] = LAYOUT(
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_PSCR,    \
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_PGUP,  \
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGDN,    \
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_END,      \
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]:
```suggestion
      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_PSCR,
      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_PGUP,
      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGDN,
      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_END,
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RCTL, MO(_RAISE),        KC_LEFT, KC_DOWN, KC_RIGHT),
+
+  [_RAISE] = LAYOUT(
+      KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,   KC_F10,  KC_F11,   KC_F12,  KC_DEL,   KC_INS,    \
+      KC_TRNS, KC_NO,   KC_UP,   KC_NO,   RGB_TOG,RGB_VAI,RGB_HUI,RGB_SAI,KC_INS, RESET,   KC_PSCR, KC_SLCK,  KC_PAUS, KC_BSLS,  KC_TRNS, \
+      KC_TRNS, KC_LEFT, KC_DOWN, KC_RIGHT,RGB_MOD,RGB_VAD,RGB_HUD,RGB_SAD,KC_NO,  KC_NO,   KC_F14,  KC_F15,   KC_INS,            KC_TRNS,   \
+      KC_LSFT,          KC_MPRV, KC_MPLY, KC_MNXT,KC_NO,  BL_TOGG,KC_NO,  KC_MUTE,KC_VOLD, KC_VOLU, KC_NO,    KC_RSFT, RGB_MOD,  KC_HOME,       \
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]:
```suggestion
      KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,   KC_F10,  KC_F11,   KC_F12,  KC_DEL,   KC_INS,
      KC_TRNS, KC_NO,   KC_UP,   KC_NO,   RGB_TOG,RGB_VAI,RGB_HUI,RGB_SAI,KC_INS, RESET,   KC_PSCR, KC_SLCK,  KC_PAUS, KC_BSLS,  KC_TRNS,
      KC_TRNS, KC_LEFT, KC_DOWN, KC_RIGHT,RGB_MOD,RGB_VAD,RGB_HUD,RGB_SAD,KC_NO,  KC_NO,   KC_F14,  KC_F15,   KC_INS,            KC_TRNS,
      KC_LSFT,          KC_MPRV, KC_MPLY, KC_MNXT,KC_NO,  BL_TOGG,KC_NO,  KC_MUTE,KC_VOLD, KC_VOLU, KC_NO,    KC_RSFT, RGB_MOD,  KC_HOME,
```

File: keyboards/wheatfield/blocked65/readme.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)  
+Hardware Supported: Wheatfield Blocked65
+Hardware Availability: TaoBao 
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make wheatfield/blocked65:default
+
+See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/readme.md]:
```suggestion
# Wheatfield Blocked65

Commonly referred to as Fanoe for Fake Canoe

The Wheatfield Blocked65 is a 65% layout PCB with arrow keys and a blocker to reduce the right-hand modifiers to 1.25u. Generally available on TaoBao.

* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
* Hardware Supported: Wheatfield Blocked65
* Hardware Availability: TaoBao

Make example for this keyboard (after setting up your build environment):

    make wheatfield/blocked65:default

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/wheatfield/blocked65/rules.mk [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = no         # Console for debug(+400)
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
```

File: keyboards/wheatfield/blocked65/rules.mk [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+BOOTLOADER = atmel-dfu
+
+# Boot Section Size in *bytes*
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
```

File: keyboards/wheatfield/blocked65/rules.mk [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+BOOTLOADER = atmel-dfu
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu
```

File: keyboards/wheatfield/blocked65/config.h (hunk 4 above)
Comment (TerryMathews) [reply keyboards/wheatfield/blocked65/config.h]: Fixed this. I'm thinking I failed to push a commit... Some of this stuff I remember fixing.

File: keyboards/wheatfield/blocked65/rules.mk [hunk 12]
```diff
@@ -0,0 +1,31 @@
+# MCU name
+MCU = atmega32u4
+
+
```
Comment (fauxpark) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
```

File: keyboards/wheatfield/blocked65/info.json [hunk 13]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "Wheat Field blocked 65% keyboard", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "bootloader": "",
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/info.json]:
```suggestion
```

File: keyboards/wheatfield/blocked65/blocked65.h [hunk 14]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "quantum.h"
+
+#define _x_ KC_NO
+
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/blocked65.h]:
```suggestion
#define LAYOUT_65_ansi_blocker( \
```

File: keyboards/wheatfield/blocked65/info.json [hunk 15]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "Wheat Field blocked 65% keyboard", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "bootloader": "", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/info.json]:
```suggestion
        "LAYOUT_65_ansi_blocker": {
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+
+enum layers {
+  _QWERTY,
+  _RAISE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_QWERTY] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]:
```suggestion
  [_QWERTY] = LAYOUT_65_ansi_blocker(
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_PSCR,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_PGUP,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGDN,
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_END,
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RCTL, MO(_RAISE),        KC_LEFT, KC_DOWN, KC_RIGHT),
+
+  [_RAISE] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]:
```suggestion
  [_RAISE] = LAYOUT_65_ansi_blocker(
```

File: keyboards/wheatfield/blocked65/readme.md [hunk 18]
```diff
@@ -0,0 +1,15 @@
+# Wheatfield Blocked65
+
+Commonly referred to as Fanoe for Fake Canoe
+
+The Wheatfield Blocked65 is a 65% layout PCB with arrow keys and a blocker to reduce the right-hand modifiers to 1.25u. Generally available on TaoBao.
+
+* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
+* Hardware Supported: Wheatfield Blocked65
+* Hardware Availability: TaoBao
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/readme.md]: There should be a direct product link here, if you have one.

File: keyboards/wheatfield/blocked65/blocked65.h [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+  K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B,      K1D, K2E, \
+  K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E, \
+  K40, K41, K42,                K46,      K48, K49, K4A, K4B, K4C, K4D       \
+){ \
+  { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E }, \
+  { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E }, \
+  { K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, _x_, K2E }, \
+  { K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E }, \
+  { K40, K41, K42, _x_, _x_, _x_, K46, _x_, K48, K49, K4A, K4B, K4C, K4D, K4E }, \
+}
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/blocked65.h]: Is this two PCBs with slightly different layouts, or one PCB that supports both?

Also, this layout macro should have a tree in `info.json` also.

File: keyboards/wheatfield/blocked65/rules.mk [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/wheatfield/blocked65/rules.mk [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+# TAP_DANCE_ENABLE = yes
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/rules.mk]: Enable [Community Layout](https://docs.qmk.fm/#/feature_layouts) support.

```suggestion

LAYOUTS = 65_ansi_blocker
```

**Comment by TerryMathews** [comment]
@fauxpark I didn't mean to force close this. I was trying to update my master and start a separate branch. Is there any way to recover this?

**Comment by noroadsleft** [comment]
@TerryMathews 

```
git fetch upstream pull/8304/head:<branch_name>
git checkout <branch_name>
```

`upstream` should be the alias for the qmk/qmk_firmware repository; run `git remote -v` to check the names. Replace `<branch_name>` with whatever you want the local branch to be named; probably `dev_branch` (though if you have a `dev_branch` locally already you'll need to rename or delete it).

**Comment by TerryMathews** [comment]
Thanks. I got frustrated and just cleaned up the code and resubmitted as #8339.

**Comment by noroadsleft** [comment]
I literally just saw your new PR.

Sorry I couldn't get to you sooner, but glad you got it sorted.

**Pull request closed without merging**